app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -200,6 +200,9 @@ impl<'a> Builder<'a> {
             .displayed_by_default
             .sort_by(|lhs, rhs| ComponentOrderingKey::of(lhs).cmp(&ComponentOrderingKey::of(rhs)));
         self.built_list
+            .components
+            .sort_by(|lhs, rhs| ComponentOrderingKey::of(lhs).cmp(&ComponentOrderingKey::of(rhs)));
+        self.built_list
```

Comment:
Nice catch, you're right. It can only be solved by storing the original components list separately and never sorting it. It is both inefficient and requires substantial refactoring to do things in the right way.